### ISSUE TITLE
Code base needs to be modernized to drop findDOMNode and string refs

### ISSUE THREAD

**Issue by stefcameron** [focus-trap/focus-trap-react#77]
The following lint issues had to be disabled in order to maintain long backward compatibility to React 16.0:

- [react/no-find-dom-node](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-find-dom-node.md): Suggested solution is to use [callback refs](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs).
- [react/no-render-return-value](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-render-return-value.md): See [deprecation warning](https://reactjs.org/docs/react-dom.html#render). Suggested solution is to use [callback refs](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs).
- [react/no-string-refs](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-string-refs.md): Suggested solution is to use [callback refs](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs).

TL/DR: Using [callback refs](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) *should* clear all these up.

Sooner or later, however, these issues will have to be rectified in order for the code to be compatible with the next major release of React, at which point, we'll have to drop support for < React 16.3 which is when `React.createRef()` and callback refs were [introduced](https://github.com/facebook/react/blob/master/CHANGELOG.md#1630-march-29-2018) (AFAICT).

**Comment by stefcameron** [comment]
No progress. Is this causing you trouble?

**Comment by m7kvqbe1** [comment]
This is causing us trouble with React.StrictMode warnings. Are there any plans to move away from `findDOMNode` soon?

![Screenshot 2022-04-27 at 15 36 50](https://user-images.githubusercontent.com/48086589/165543678-2dda8b02-6e0e-44f4-90d8-9c4143962fa7.png)

**Comment by stefcameron** [comment]
@m7kvqbe1 Thanks for letting me know. As it stands, this would be the next priority to work on for this project, provided nothing more urgent arises in the meantime. I can't make promises on a timeline, however. Just hopefully "soon". Issues on focus-trap and tabbable could also detract from this, but as of this past Thursday, I _think_ I finally got all 3 in a good place again. 😅

**Comment by stefcameron** [comment]
Hoping #710 will fix this.

**Comment by stefcameron** [comment]
Look for a fix to this (I hope!!) in v9.0.0 being published right now.
